src/libcore/fmt/mod.rs

```diff
@@ -166,7 +166,9 @@ pub struct Formatter<'a> {
 // NB. Argument is essentially an optimized partially applied formatting function,
 // equivalent to `exists T.(&T, fn(&T, &mut Formatter) -> Result`.
 
-enum Void {}
+struct Void {
+    _private: (),
+}
```

Comment:
What's the problem with this change? I think this is what @nikomatsakis meant by redefining `Void`. We need to make *some* change here or else that module breaks.